packages/react/src/keyframes/__tests__/index.test.tsx

```diff
@@ -0,0 +1,324 @@
+import { keyframes, styled } from '@compiled/react';
+import { render } from '@testing-library/react';
+
+import defaultFadeOut, { fadeOut as shadowedFadeOut, namedFadeOut } from '../__fixtures__';
+
+const getOpacity = (str: string | number) => str;
+
+const getKeyframe = (name: string) => {
+  const styles = Array.from(
+    document.body.querySelectorAll('style'),
+    (style) => style.innerHTML
+  ).join('\n');
+
+  return styles.substring(styles.indexOf(`@keyframes ${name}`));
+};
+
+describe('keyframes', () => {
+  describe('referenced through a css prop', () => {
+    describe('render an animation', () => {
+      it('given an object call expression argument', () => {
+        const fadeOut = keyframes({
+          from: {
+            opacity: 1,
+          },
+          to: {
+            opacity: 0,
+          },
+        });
+
+        const { getByText } = render(<div css={{ animationName: fadeOut }}>hello world</div>);
+
+        expect(getByText('hello world')).toHaveCompiledCss('animation-name', 'k1m8j3od');
+        expect(getKeyframe('k1m8j3od')).toMatchInlineSnapshot(
+          `"@keyframes k1m8j3od{0%{opacity:1}to{opacity:0}}"`
+        );
+      });
+
+      it('given a template literal call expression argument', () => {
+        const fadeOut = keyframes(`from { opacity: 1; } to { opacity: 0; }`);
+        const { getByText } = render(<div css={{ animationName: fadeOut }}>hello world</div>);
+
+        expect(getByText('hello world')).toHaveCompiledCss('animation-name', 'klmf72q');
+        expect(getKeyframe('klmf72q')).toMatchInlineSnapshot(
+          `"@keyframes klmf72q{0%{opacity:1}to{opacity:0}}"`
```

Comment:
Hey @MonicaOlejniczak, most likely I am overthinking, but I would expect same keyframes to generate the same output. All these tests have the same properties `{0%{opacity:1}to{opacity:0}}`, but generate different names.

Maybe we could check how other libraries deal with keyframes

Reply:
I did initially as well, but it's because I am generating the name from the raw input before processing the css. So a keyframe of `keyframes({})` with the same values, will be different to `keyframes('')`. This should be fine in practice, as applications should use a consistent syntax.

Emotion and styled components both generate the name by hashing the resulting CSS, which could be done as an optimisation later but there is a bit of trickiness to it.